### PR TITLE
[QOL-7841] enable SQL functions needed for the ICT dashboard

### PR DESCRIPTION
**Pull request by ThrawnCA**

### CODE REVIEW

File: files/default/allowed_functions.txt [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+variance
+width
+width_bucket
+xmlagg
+xmlcomment
+xmlconcat
+xpath
+xpath_exists
+COUNT
+ictdashboards_tables
```
Comment (duttonw) [inline files/default/allowed_functions.txt:284]: can you also add capitalized version of this. ICTDASHBOARDS_TABLES as the whitelist is case sensitive

**Review by duttonw — APPROVED** [review]

File: files/default/allowed_functions.txt (hunk 1 above)
Comment (duttonw) [reply files/default/allowed_functions.txt:284]: can it take comments # something and still work?

Comment (ThrawnCA) [reply files/default/allowed_functions.txt:284]: I'm raising a pull request on CKAN core to make the whitelist case-insensitive. For now, I'm pretty sure the ICT dashboard only needs the lowercase version.

Yes, it looks like comments are allowed.

File: templates/default/ckan_properties.ini.erb [hunk 2]
```diff
@@ -73,6 +73,10 @@ ckan.datastore.read_url = postgresql://<%= node['datashades']['ckan_web']['dsuse
 # See https://github.com/ckan/ckanext-xloader/issues/17 and https://github.com/ckan/ckanext-xloader/issues/83
 ckanext.xloader.jobs_db.uri = postgresql://<%= node['datashades']['ckan_web']['dbuser'] %>:<%= node['datashades']['postgres']['password'] %>@<%= node['datashades']['version'] %>pg.<%= node['datashades']['tld'] %>/<%= node['datashades']['ckan_web']['dbname'] %>
 
+# We need the SQL endpoint, despite the non-ideal security implications
+ckan.datastore.sqlsearch.enabled = True
+ckan.datastore.sqlsearch.allowed_functions_file = %(here)s/allowed_functions.txt
```
Comment (duttonw) [inline templates/default/ckan_properties.ini.erb:78]: %(here)s does what?

**Review by duttonw — APPROVED** [review]

File: templates/default/ckan_properties.ini.erb (hunk 2 above)
Comment (ThrawnCA) [reply templates/default/ckan_properties.ini.erb:78]: It appears to be the directory containing this file. See line 62 for an example.

**Pull request merged**